src/outputs/restart_hdf5.cpp

```diff
@@ -180,64 +181,28 @@ void RestartReaderHDF5::ReadParams(const std::string &name, Params &p) {
 #endif // ENABLE_HDF5
 }
 void RestartReaderHDF5::ReadBlocks(const std::string &name, IndexRange range,
+                                   const OutputUtils::VarInfo &info,
                                    std::vector<Real> &dataVec,
-                                   const std::vector<size_t> &bsize,
-                                   int file_output_format_version, MetadataFlag where,
-                                   const std::vector<int> &shape) const {
+                                   int file_output_format_version) const {
 #ifndef ENABLE_HDF5
   PARTHENON_FAIL("Restart functionality is not available because HDF5 is disabled");
 #else  // HDF5 enabled
   auto hdl = OpenDataset<Real>(name);
 
-  constexpr int CHUNK_MAX_DIM = 7;
+  const int VNDIM = info.VNDIM;
 
   /** Select hyperslab in dataset **/
-  hsize_t offset[CHUNK_MAX_DIM] = {static_cast<hsize_t>(range.s), 0, 0, 0, 0, 0, 0};
-  hsize_t count[CHUNK_MAX_DIM];
   int total_dim = 0;
-  if (file_output_format_version == -1) {
-    size_t vlen = 1;
-    for (int i = 0; i < shape.size(); i++) {
-      vlen *= shape[i];
-    }
-    count[0] = static_cast<hsize_t>(range.e - range.s + 1);
-    count[1] = bsize[2];
-    count[2] = bsize[1];
-    count[3] = bsize[0];
-    count[4] = vlen;
-    total_dim = 5;
-  } else if (file_output_format_version == 2) {
-    PARTHENON_REQUIRE(shape.size() <= 1,
-                      "Higher than vector datatypes are unstable in output versions < 3");
-    size_t vlen = 1;
-    for (int i = 0; i < shape.size(); i++) {
-      vlen *= shape[i];
-    }
-    count[0] = static_cast<hsize_t>(range.e - range.s + 1);
-    count[1] = vlen;
-    count[2] = bsize[2];
-    count[3] = bsize[1];
-    count[4] = bsize[0];
-    total_dim = 5;
-  } else if (file_output_format_version == HDF5::OUTPUT_VERSION_FORMAT) {
-    count[0] = static_cast<hsize_t>(range.e - range.s + 1);
-    const int ndim = shape.size();
-    if (where == MetadataFlag(Metadata::Cell)) {
-      for (int i = 0; i < ndim; i++) {
-        count[1 + i] = shape[ndim - i - 1];
-      }
-      count[ndim + 1] = bsize[2];
-      count[ndim + 2] = bsize[1];
-      count[ndim + 3] = bsize[0];
-      total_dim = 3 + ndim + 1;
-    } else if (where == MetadataFlag(Metadata::None)) {
-      for (int i = 0; i < ndim; i++) {
-        count[1 + i] = shape[ndim - i - 1];
-      }
-      total_dim = ndim + 1;
-    } else {
-      PARTHENON_THROW("Only Cell and None locations supported!");
-    }
+  hsize_t offset[VNDIM], count[VNDIM];
+  std::fill(offset + 1, offset + VNDIM, 0);
+  std::fill(count + 1, count + VNDIM, 1);
+
+  offset[0] = static_cast<hsize_t>(range.s);
+  count[0] = static_cast<hsize_t>(range.e - range.s + 1);
+  const IndexDomain domain = hasGhost ? IndexDomain::entire : IndexDomain::interior;
+
+  if (file_output_format_version >= HDF5::OUTPUT_VERSION_FORMAT - 1) {
```

Comment:
Maybe add a comment that this is hardcoded logic supporting restarts from 3 with current 4? (or be more specific about the numbers so that it's clear it needs an update/care when the format is changed again -- which will hopefully never happen)

Reply:
Comment added.